controllers/xsite.go

```diff
@@ -43,18 +39,64 @@ func (r *infinispanRequest) applyLabelsToCoordinatorsPod(podList *corev1.PodList
 					crossSiteViewFormed.Message = fmt.Sprintf("Error: %s", err.Error())
 				}
 				return crossSiteViewFormed, nil
-			} else {
-				if ok && lab == strconv.FormatBool(ok) {
-					// If present leave the label but false the value
-					if ok {
-						item.Labels[consts.CoordinatorPodLabel] = strconv.FormatBool(cacheManagerInfo.Coordinator)
-						if err = r.Client.Update(r.ctx, &item); err != nil {
-							return nil, err
-						}
-					}
-				}
 			}
 		}
 	}
 	return &ispnv1.InfinispanCondition{Type: ispnv1.ConditionCrossSiteViewFormed, Status: metav1.ConditionFalse, Message: "Coordinator not ready"}, nil
 }
+
+// GetGossipRouterDeployment returns the deployment for the Gossip Router pod
+func (r *infinispanRequest) GetGossipRouterDeployment(m *infinispanv1.Infinispan) *appsv1.Deployment {
+	name := m.GetGossipRouterDeploymentName()
+	lsTunnel := GossipRouterPodLabels(m.Name)
+	replicas := int32(1)
+
+	// if the user configures 0 replicas, shutdown the gossip router pod too.
+	if m.Spec.Replicas <= 0 {
+		replicas = 0
+	}
+
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
```

Comment:
Probably we need  a specific name for GossipRouter deployment, not the same as cluster name, i.e. with `tunnel` prefix or postfix

Reply:
name is `name := m.GetGossipRouterDeploymentName()`
`-tunnel` is added to the cluster name.

Reply:
actually, I don't need the variable `name`. I'll remove it to avoid confusion